### DISCUSSION TITLE
Atom perrsistance usage with multiple stores / Providers

### DISCUSSION THREAD

**Discussion by lsironic** [pmndrs/jotai#2881] [Q&A]
if we use atomWithStorage within multiple providers, only the last value is persisted.
Is there a way to perisit each value by Provider?

**Comment by dai-shi** [comment]
It's a good question. I'd say it's limited by design. We need multiple keys to store each value, but atoms don't know about stores.

For now, only a workaround seems to create multiple atoms for each provider.
